presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -236,8 +236,10 @@ public PlanNode visitGroupId(GroupIdNode node, RewriteContext<Expression> contex
         @Override
         public PlanNode visitMarkDistinct(MarkDistinctNode node, RewriteContext<Expression> context)
         {
-            checkState(!SymbolsExtractor.extractUnique(context.get()).contains(node.getMarkerSymbol()), "predicate depends on marker symbol");
-            return context.defaultRewrite(node, context.get());
+            if (!SymbolsExtractor.extractUnique(context.get()).contains(node.getMarkerSymbol())) {
```

Comment:
Predicates can be pushed only if they contain symbols from `MarkDistinctNode#distinctSymbols`.